mm2src/mm2_main/src/lp_swap/maker_swap.rs

```diff
@@ -630,7 +650,25 @@ impl MakerSwap {
             },
         };
         drop(send_abort_handle);
-        let taker_fee = match self.taker_coin.tx_enum_from_bytes(&payload) {
+        let mut swap_events = vec![];
+        if let Some(instructions) = payload.instructions() {
+            match self
+                .maker_coin
+                .validate_instructions(instructions, &self.secret_hash(), self.maker_amount.clone())
```

Comment:
Does `validate_instructions` return `Ok(None)` if the coin doesn't expect them? How about returning an error in this case?